core/src/test-core/java/com/radixdlt/environment/NoEpochsConsensusModule.java

```diff
@@ -64,42 +64,256 @@
 public class NoEpochsConsensusModule extends AbstractModule {
+
   @Override
   public void configure() {
+    bind(PacemakerState.class).in(Scopes.SINGLETON);
+    bind(PacemakerReducer.class).to(PacemakerState.class);
+    bind(ExponentialPacemakerTimeoutCalculator.class).in(Scopes.SINGLETON);
+    bind(PacemakerTimeoutCalculator.class).to(ExponentialPacemakerTimeoutCalculator.class);
+
     OptionalBinder.newOptionalBinder(
         binder(), EpochManager.class); // So that this is consistent with tests
     var eventBinder =
         Multibinder.newSetBinder(binder(), new TypeLiteral<Class<?>>() {}, LocalEvents.class)
             .permitDuplicates();
+
     eventBinder.addBinding().toInstance(TimeoutQuorumDelayedResolution.class);
+    eventBinder.addBinding().toInstance(RoundUpdate.class);
+    eventBinder.addBinding().toInstance(BFTRebuildUpdate.class);
+    eventBinder.addBinding().toInstance(BFTInsertUpdate.class);
+    eventBinder.addBinding().toInstance(Proposal.class);
+    eventBinder.addBinding().toInstance(Vote.class);
+    eventBinder.addBinding().toInstance(LedgerUpdate.class);
     eventBinder.addBinding().toInstance(ScheduledLocalTimeout.class);
     eventBinder.addBinding().toInstance(VertexRequestTimeout.class);
     eventBinder.addBinding().toInstance(ProposalRejected.class);
-    eventBinder.addBinding().toInstance(RoundUpdate.class);
-    eventBinder.addBinding().toInstance(LedgerUpdate.class);
+  }
+
+  @Provides
+  @Singleton
+  public VertexStoreBFTSyncRequestProcessor syncRequestProcessor(
+      VertexStoreAdapter vertexStore,
+      RemoteEventDispatcher<NodeId, GetVerticesErrorResponse> errorResponseDispatcher,
+      RemoteEventDispatcher<NodeId, GetVerticesResponse> responseDispatcher,
+      Metrics metrics) {
+    return new VertexStoreBFTSyncRequestProcessor(
+        vertexStore, errorResponseDispatcher, responseDispatcher, metrics);
+  }
+
+  @Provides
+  @Singleton
+  public ProposerElection proposerElection(BFTConfiguration configuration) {
+    return configuration.getProposerElection();
+  }
+
+  @Provides
+  @Singleton
+  public BFTEventProcessor bftEventProcessor(
+      @Self SelfValidatorInfo self,
+      BFTConfiguration config,
+      Pacemaker pacemaker,
+      BFTSync bftSync,
+      SafetyRules safetyRules,
+      Hasher hasher,
+      HashVerifier verifier,
+      TimeSupplier timeSupplier,
+      ProposerElection proposerElection,
+      Metrics metrics,
+      EventDispatcher<RoundQuorumResolution> roundQuorumResolutionEventDispatcher,
+      ScheduledEventDispatcher<TimeoutQuorumDelayedResolution>
+          timeoutQuorumDelayedResolutionDispatcher,
+      EventDispatcher<ConsensusByzantineEvent> doubleVoteEventDispatcher,
+      EventDispatcher<ProposalRejected> proposalRejectedDispatcher,
+      RoundUpdate roundUpdate,
+      @TimeoutQuorumResolutionDelayMs long timeoutQuorumResolutionDelayMs) {
+    /*
+    TODO: consider cleaning this up (but most probably it's not worth it :))
```

Comment:
Out of interest, would you support ditching this and instead just using really long epochs? This part of the code always fills me with a sadness that we're testing mocks and things are configured in unrealistic and flaky ways.

Do you think supporting this provides us any benefits worth keeping it around for?

(Not something to address in this PR of course!)

Reply:
That's an interesting idea, definitely something to consider